console/v1alpha1/types_console_plugin.go

```diff
@@ -27,16 +27,49 @@ type ConsolePluginSpec struct {
 	// service serving certificate. The console backend will proxy the
 	// plugins assets from the Service using the service CA bundle.
 	Service ConsolePluginService `json:"service"`
+	// proxy is a list of services that the plugin needs to connect to.
+	Proxy ConsolePluginProxy `json:"proxy"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// services is list of Kubernetes Service to which the plugin
+	// will connect to, using a deployment with an HTTP server.
+	// The Service must use HTTPS and service serving certificate.
+	// The console backend will proxy the communication between
+	// plugin and the specified Service.
+	Services []ConsolePluginProxyService `json:"services"`
+}
+
+// ConsolePluginProxyService holds information on service to which
+// console's backend will proxy the plugin's requests.
+type ConsolePluginProxyService struct {
+	// name of Service that the plugin needs to connect to.
+	Name string `json:"name"`
+	// namespace of Service that the plugin needs to connect to
+	Namespace string `json:"namespace"`
+	// port on which the Service that the plugin needs to connect to
+	// is listening on.
+	Port int32 `json:"port"`
+	// caCertificate provides the cert authority certificate contents,
+	// in case the proxied service is using custom service CA.
+	// By default service CA bundle is used.
+	CACertificate string `json:"caCertificate"`
+	// authorize indicates if the proxied request will contain
+	// user's OpenShift access token. By default the access token
+	// is not part of the proxied request.
```

Comment:
You should mention how the token is passed in the `Authorization` request header.